.github/workflows/osrm-backend.yml

```diff
@@ -212,6 +212,17 @@ jobs:
             CXXCOMPILER: clang++-6.0
             ENABLE_CONAN: ON
 
+          - name: gcc-12-release
+            continue-on-error: false
+            node: 16
+            runs-on: ubuntu-22.04
+            BUILD_TOOLS: ON
+            BUILD_TYPE: Release
+            CCOMPILER: gcc-12
+            CXXCOMPILER: g++-12
+            ENABLE_BENCHMARKS: ON
+            CXXFLAGS: '-Wno-array-bounds -Wno-uninitialized'
```

Comment:
Had to disable a couple of warnings as some of thirdparty dependencies(sol2 in particular) don't compile otherwise - probably we have some misconfiguration as in general warnings in third party deps shouldn't bother us, but for now it is easier option(at least for now we have some build with gcc 12 and afterwards we can try to understand what is wrong with sol2)